File: src/main/java/pl/tsimur/jpatest/model/Organization.java
package pl.tsimur.jpatest.model;


import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table
public class Organization {
    @Id
    private Integer nip;

    private String name;

    private String address;

    @OneToMany(mappedBy = "employer", fetch = FetchType.LAZY)
    @JsonIgnore
    private List<User> employees;

    @OneToOne
    @JoinColumn(name = "owner_id")
    @JsonIgnore
    private User owner;
}

File: src/main/kotlin/pl/timur/jpatest/model/Organization.kt
package pl.timur.jpatest.model

import com.fasterxml.jackson.annotation.JsonIgnore
import javax.persistence.*

@Entity
@Table
data class Organization(
    @Id
    val nip: Int,
    val name: String,
    val address: String
) {
    @OneToMany(mappedBy = "employer", fetch = FetchType.LAZY)
    @JsonIgnore
    val employees: List<User> = ArrayList()

    @OneToOne
    @JoinColumn(name = "owner_id")
    @JsonIgnore
    val owner: User? = null
}